stumpy/stumpi.py

```diff
@@ -124,10 +124,25 @@ def __init__(self, T, m, egress=True, normalize=True, p=2.0):
 
         self._T, self._M_T, self._Σ_T = core.preprocess(self._T, self._m)
         # Retrieve the left matrix profile values
-        for i, j in enumerate(self._left_I):
-            if j >= 0:
-                D = core.mass(self._T[i : i + self._m], self._T[j : j + self._m])
-                self._left_P[i] = D[0]
+
+        # if left neighbor of `S[i] = T[i, i+m]` is actually its matrix profile index
+        # neighbor, then their distance is matrix profile[i]!!!
+        mask = self._left_I == self._I
+        self._left_P[mask] = self._P[mask]
+
+        # the remaining ones
```

Comment:
This is not a good comment. Maybe:

"# Only recompute the `i`th left matrix profile value, `self._left_P[i]`,  when `self._P[i] != self._left_P[i]`"

Reply:
@seanlaw 
Sorry about that. I should have put time into writing informative comments. I will revise it.

Reply:
How about this?
"# Only recompute the ith left matrix profile value, self._left_P[i], when self._I[i] != self._left_I[i]"

(for the same reason I explained above)

Reply:
Go for it!